pkgs/build-support/node/fetch-yarn-deps/index.js

```diff
@@ -20,15 +21,66 @@ const exec = async (...args) => {
 	return res
 }
 
-const downloadFileHttps = (fileName, url, expectedHash, hashType = 'sha1') => {
+const createURLProxyMap = () => {
+	let proxyMap = new Map();
+
+	const proxiesForHttp = [];
+	const proxiesForHttps = [];
+
+	if (process.env.HTTP_PROXY) {
+		console.log('Detected HTTP_PROXY');
+		proxiesForHttp.push(process.env.HTTP_PROXY);
+	}
+	if (process.env.http_proxy) {
+		console.log('Detected http_proxy');
+		proxiesForHttp.push(process.env.http_proxy);
+	}
+	if (process.env.HTTPS_PROXY) {
+		console.log('Detected HTTPS_PROXY');
+		proxiesForHttps.push(process.env.HTTPS_PROXY);
+	}
+	if (process.env.https_proxy) {
+		console.log('Detected https_proxy');
+		proxiesForHttps.push(process.env.https_proxy);
+	}
+	if (process.env.all_proxy) {
+		console.log('Detected all_proxy');
+		proxiesForHttp.push(process.env.all_proxy);
+		proxiesForHttps.push(process.env.all_proxy);
+	}
+	if (process.env.ALL_PROXY) {
+		console.log('Detected ALL_PROXY');
+		proxiesForHttp.push(process.env.ALL_PROXY);
+		proxiesForHttps.push(process.env.ALL_PROXY);
+	}
+	proxyMap.set('http:', proxiesForHttp);
+	proxyMap.set('https:', proxiesForHttps);
+	return proxyMap;
+}
+
+const URL_PROXY_MAP = createURLProxyMap();
+
+const getProxyForUrl = (url) => {
+	const urlParsed = new URL(url);
+	if (URL_PROXY_MAP.has(urlParsed.protocol)) {
+		let proxiesForProtocol = URL_PROXY_MAP.get(urlParsed.protocol);
+		if (proxiesForProtocol.length > 0) {
+			let proxyForUrl = proxiesForProtocol[0];
+			console.log(`Selecting proxy ${proxyForUrl} for ${url}`);
+			return proxyForUrl;
+		}
+	}
+	console.log(`No proxy selected for ${url}`);
+	return '';
+}
+
+const getInner = function (url, agent, fileName, expectedHash, hashType) {
 	return new Promise((resolve, reject) => {
-		const get = (url, redirects = 0) => https.get(url, (res) => {
-			if(redirects > 10) {
-				reject('Too many redirects!');
-				return;
-			}
+		https.get(url, {
+			agent: agent,
+		}, (res) => {
 			if(res.statusCode === 301 || res.statusCode === 302) {
-				return get(res.headers.location, redirects + 1)
```

Comment:
Could it be that this change can be decoupled from the main functionality addition this PR suggests?